diesel_derives/src/identifiable.rs

```diff
@@ -21,9 +23,19 @@ pub fn derive(item: DeriveInput) -> Result<TokenStream> {
     let mut field_ty = Vec::new();
     let mut field_name = Vec::new();
     for pk in model.primary_key_names.iter() {
-        let f = model.find_column(pk)?;
-        field_ty.push(&f.ty);
-        field_name.push(&f.name);
+        let Field {
+            ty,
+            name,
+            serialize_as,
+            ..
+        } = &model.find_column(pk)?;
+        if let Some(AttributeSpanWrapper { item: ty, .. }) = serialize_as.as_ref() {
+            field_ty.push(quote!(#ty));
```

Comment:
I believe that might be seen as breaking change, although one could also argue that this is a bug fix. Pinging @diesel-rs/contributors to get some more opinions on this.

Reply:
(I'm a recent Diesel reviewer, so don't give my opinion too much weight)
There are two changes here, right?

- One seems to me clearly a bugfix: for a composite primary key, we miss the `&'ident` / `&self` for all but the first element, leading to potential nonsense. We should just fix that.
- The other is less clear: we apply `serialize_as` when we didn't before. Clearly we "should" apply it, but could people be depending on this in some way? I don't have enough experience to know.

Reply:
As far as I know, composite primary key already works (as shown in the tests), so this is not a fix. The `&'ident` / `&self` part was repeated for every element already, as per [`quote!`'s interpolation](https://docs.rs/quote/1.0.9/quote/macro.quote.html#interpolation).

Here I changed `field_ty` and `field_name` to be `Vec<TokenStream>` directly, so I can push what will be needed by the `impl` directly.